pkg/api/routes.go

```diff
@@ -514,7 +514,7 @@ func (rh *RouteHandler) GetManifest(response http.ResponseWriter, request *http.
 
 	if rh.c.MetaDB != nil {
 		err := meta.OnGetManifest(name, reference, mediaType, content, rh.c.StoreController, rh.c.MetaDB, rh.c.Log)
-		if err != nil {
```

Comment:
This seems too wide of a net for me, and metadb errors with other root causes may go unnoticed (not even logged afaik). Do we want this? On the bright side clients would have a functioning API even in those cases.

As an alternative we could update `storage.CheckIsImageSignature` to also look for annotations. I see cosign adds the "dev.cosignproject.cosign/signature" annotation to all the layers in the manifest. Or later mediaType "application/vnd.dev.cosign.simplesigning.v1+json"

Reply:
We don't want to pollute zot with all unknown types. The case is we found the manifest (correctly) from the local storage, we didn't find it in the metadb since we didn't put it there, so do you want to fail the request for the specific error of "not found"?

Reply:
This is a cosign specific signature miss-identification, I think we should fix the cosign specify logic it there in the cosign specific logic we already have.

We could also keep the generic error handling as in this PR if returning the content is more important compared to having accurate download counters.

So, is the following scenario working correctly?
1. Push container image by tag (regular image, no special artifact cases).
2. Pull container image by digest.

Reply:
```
 skopeo copy --format=oci --preserve-digests --dest-tls-verify=false --dest-creds=test:test docker://ghcr.io/project-zot/zot-linux-amd64@sha256:c4d8e94e78e340ac7a57d80becca26bce4d2fbd1233cb6f9280e02d100820a17 docker://localhost:5000/ghcr.io/project-zot/zot-linux-amd64@sha256:c4d8e94e78e340ac7a57d80becca26bce4d2fbd1233cb6f9280e02d100820a17
 
 oras manifest fetch localhost:5000/ghcr.io/project-zot/zot-linux-amd64@sha256:c4d8e94e78e340ac7a57d80becca26bce4d2fbd1233cb6f9280e02d100820a17
 ```
 
 ^ appears to work correctly.

Reply:
Per discussion best to take the approach in this PR.

Reply:
Added a test case.